amd64/root/firmware-update.sh

```diff
@@ -156,11 +160,13 @@ if [[ ! -f ${FLASHER_PARAM_VALUES[-f]} ]]; then
     echo " "
     echo "Downloading..."
     echo " "
-    curl --fail --output "${FLASHER_PARAM_VALUES[-f]}" "${FW_ONLINE_BASE%/}/$fileName" && [[ -f ${FLASHER_PARAM_VALUES[-f]} ]]
-    delete_and_exit_on_error "${FLASHER_PARAM_VALUES[-f]}" "Download Error! Please re-run this script..."
+    curl --fail --silent --output "${FLASHER_PARAM_VALUES[-f]}" "${FW_ONLINE_STABLE_BASE%/}/$fileName" || \
```

Comment:
On entering the if-block in case the file doesn't exist locally i would:
1. Loop over possible bases and check for the fw:
```shell
for base in "${FW_ONLINE_BASES[@]}"; do
    fw_url="${base%/}/$fileName"
    curl --fail --silent --head --output /dev/null "$fw_url" && break
done
exit_on_error "Can't find '$filename' neither locally nor online. Exiting ..."
```
2. State that the file doesn't exist and ask if if should be downloaded from`$fw_url`.